tests/lib/rules/object-curly-newline.js

```diff
@@ -659,6 +764,242 @@ ruleTester.run("object-curly-newline", rule, {
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                { line: 1, column: 5, message: "Expected a line break after this opening brace." },
+                { line: 2, column: 2, message: "Expected a line break before this closing brace." }
+            ]
+        },
+        {
+            code: [
+                "let {e = function() {",
+                "dosomething();",
+                "}} = a;"
+            ].join("\n"),
+            output: [
+                "let {",
+                "e = function() {",
+                "dosomething();",
+                "}",
+                "} = a;"
+            ].join("\n"),
+            options: [{ multiline: true, consistent: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                { line: 1, column: 5, message: "Expected a line break after this opening brace." },
+                { line: 3, column: 2, message: "Expected a line break before this closing brace." }
+            ]
+        },
+        {
+            code: [
+                "var a = {a: 1, b: 2};"
+            ].join("\n"),
+            output: [
+                "var a = {",
+                "a: 1, b: 2",
+                "};"
+            ].join("\n"),
+            options: [{ multiline: true, minProperties: 2, consistent: true }],
+            errors: [
+                { line: 1, column: 9, message: "Expected a line break after this opening brace." },
+                { line: 1, column: 20, message: "Expected a line break before this closing brace." }
+            ]
+        },
+
+        // "consistent" and "minProperties" ------------------------------------------
+        {
+            code: [
+                "var c = {a: 1, b: 2};"
+            ].join("\n"),
+            output: [
+                "var c = {",
+                "a: 1, b: 2",
+                "};"
+            ].join("\n"),
+            options: [{ multiline: true, consistent: true, minProperties: 2 }],
+            errors: [
+                { line: 1, column: 9, message: "Expected a line break after this opening brace." },
+                { line: 1, column: 20, message: "Expected a line break before this closing brace." }
+            ]
+        },
+        {
+            code: [
+                "let {a, b} = {",
+                "a: 1, b: 2",
+                "};"
+            ].join("\n"),
+            output: [
+                "let {",
+                "a, b",
+                "} = {",
+                "a: 1, b: 2",
+                "};"
+            ].join("\n"),
+            options: [{ multiline: true, consistent: true, minProperties: 2 }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                { line: 1, column: 5, message: "Expected a line break after this opening brace." },
+                { line: 1, column: 10, message: "Expected a line break before this closing brace." }
+            ]
+        },
+
```

Comment:
Could we also add some tests for `{ multiline: false, consistent: true }` as well?